packages/editor-ui/src/mixins/history.ts

```diff
@@ -0,0 +1,113 @@
+	},
+	destroyed() {
+		document.removeEventListener('keydown', this.handleKeyDown);
+	},
+	methods: {
+		handleKeyDown(event: KeyboardEvent) {
+			if (event.repeat) return;
+			if (this.isCtrlKeyPressed(event) && event.key === 'z') {
+				event.preventDefault();
+				if (!this.isNDVOpen) {
+					if (event.shiftKey) {
+						this.callDebounced('redo', { debounceTime: UNDO_REDO_DEBOUNCE_INTERVAL, trailing: true  });
+					} else {
+						this.callDebounced('undo', { debounceTime: UNDO_REDO_DEBOUNCE_INTERVAL, trailing: true  });
+					}
+				} else if (!event.shiftKey) {
+					this.trackUndoAttempt(event);
+				}
+			}
+		},
+		async undo() {
+			const command = this.historyStore.popUndoableToUndo();
+			if (!command) {
+				return;
+			}
+			if (command instanceof BulkCommand) {
+				this.historyStore.bulkInProgress = true;
+				const commands = command.commands;
+				const reverseCommands: Command[] = [];
+				for (let i = commands.length - 1; i >= 0; i--) {
+					await commands[i].revert();
+					reverseCommands.push(commands[i].getReverseCommand());
+				}
+				this.historyStore.pushUndoableToRedo(new BulkCommand(reverseCommands));
+				await Vue.nextTick();
+				this.historyStore.bulkInProgress = false;
+			}
+			if (command instanceof Command) {
+				await command.revert();
+				this.historyStore.pushUndoableToRedo(command.getReverseCommand());
+				this.uiStore.stateIsDirty = true;
+			}
+			this.trackCommand(command, 'undo');
+		},
+		async redo() {
+			const command = this.historyStore.popUndoableToRedo();
+			if (!command) {
+				return;
+			}
+			if (command instanceof BulkCommand) {
+				this.historyStore.bulkInProgress = true;
+				const commands = command.commands;
+				const reverseCommands = [];
+				for (let i = commands.length - 1; i >= 0; i--) {
+					await commands[i].revert();
+					reverseCommands.push(commands[i].getReverseCommand());
+				}
+				this.historyStore.pushBulkCommandToUndo(new BulkCommand(reverseCommands), false);
+				await Vue.nextTick();
+				this.historyStore.bulkInProgress = false;
+			}
+			if (command instanceof Command) {
+				await command.revert();
+				this.historyStore.pushCommandToUndo(command.getReverseCommand(), false);
+				this.uiStore.stateIsDirty = true;
+			}
+			this.trackCommand(command, 'redo');
+		},
+		trackCommand(command: Undoable, type: 'undo'|'redo'): void {
+			if (command instanceof Command) {
+				this.$telemetry.track(`User hit ${type}`, { commands_length: 1, commands: [ command.name ] });
+			} else if (command instanceof BulkCommand) {
+				this.$telemetry.track(`User hit ${type}`, { commands_length: command.commands.length, commands: command.commands.map(c => c.name) });
+			}
+		},
+		trackUndoAttempt(event: KeyboardEvent) {
+			if (this.isNDVOpen && !event.shiftKey) {
+				const activeNode = this.ndvStore.activeNode;
+				if (activeNode) {
+					this.$telemetry.track(`User hit undo in NDV`, { node_type: activeNode.type });
```

Comment:
weird we are not tracking the event itself

Reply:
Talked with Nik about it, looks like we are only interested in the fact that users attempted undo here.